pkg/common-controller/snapshot_controller.go

```diff
@@ -943,6 +943,18 @@ func (ctrl *csiSnapshotCommonController) updateSnapshotStatus(snapshot *crdv1.Vo
 	if content.Status != nil && content.Status.ReadyToUse != nil {
 		readyToUse = *content.Status.ReadyToUse
 	}
+	var volumeSnapshotErr *crdv1.VolumeSnapshotError
+	if content.Status != nil && content.Status.Error != nil {
```

Comment:
To use this change for #333 we need to clear the error in the volume snapshot object, if the content has been created or bound successfully.
```
volumeSnapshotErr := &crdv1.VolumeSnapshotError{}
if content.status != nil && content.Status.Error != nil {
   // update the error time and message
}
```

Thoughts @xing-yang ?

Reply:
Yes, make sense.,

Reply:
hmmm, isn't the following sufficient?
volumeSnapshotErr = content.Status.Error.DeepCopy()
I'd say just do it in the code blocks where assignment really happens and delete this whole block. WDYT?